services/git-service-impl/src/main/java/io/fabric8/launcher/service/bitbucket/impl/BitbucketServiceFactoryImpl.java

```diff
@@ -0,0 +1,39 @@
+package io.fabric8.launcher.service.bitbucket.impl;
+
+import static io.fabric8.launcher.service.bitbucket.api.BitbucketEnvVarSysPropNames.LAUNCHER_MISSIONCONTROL_BITBUCKET_APPLICATION_PASSWORD;
+import static io.fabric8.launcher.service.bitbucket.api.BitbucketEnvVarSysPropNames.LAUNCHER_MISSIONCONTROL_BITBUCKET_USERNAME;
+
+import java.util.Optional;
+
+import io.fabric8.launcher.base.EnvironmentSupport;
+import io.fabric8.launcher.base.identity.Identity;
+import io.fabric8.launcher.base.identity.IdentityFactory;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import io.fabric8.launcher.base.identity.UserPasswordIdentity;
+import io.fabric8.launcher.service.bitbucket.api.BitbucketService;
+import io.fabric8.launcher.service.bitbucket.api.BitbucketServiceFactory;
+
+public class BitbucketServiceFactoryImpl implements BitbucketServiceFactory {
+    @Override
+    public BitbucketService create(final Identity identity) {
+        if (!(identity instanceof UserPasswordIdentity)) {
+            throw new IllegalArgumentException("BitbucketService supports only TokenIdentity. Not supported:" + identity);
```

Comment:
This error message does not look right. Or maybe the if statement is wrong

Reply:
Good catch :) it's only UserPasswordIdentity